# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/workers/WorkerScope.cpp
# Commit: 26d3176a00ef
# Full Hash: 26d3176a00ef338cef46f4ad6f8bdfb91f2f586c
# Author: Jens Stutte <jstutte@mozilla.com>
# Date: 2021-12-08 03:42:13
# Regressor Bug: 1744025
# File Overlap Count: 3
# Description:
#   Bug 1744025: Use CheckedUnsafePtr<WorkerPrivate> in WorkerGlobalScopeBase. r=dom-worker-reviewers,asuth
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D132709
# ==============================================================================

diff -r 2c8941fdcee5 -r 26d3176a00ef dom/workers/WorkerScope.cpp
--- a/dom/workers/WorkerScope.cpp	Tue Dec 07 16:47:14 2021 +0000
+++ b/dom/workers/WorkerScope.cpp	Tue Dec 07 16:47:15 2021 +0000
@@ -209,8 +209,7 @@
 NS_INTERFACE_MAP_END_INHERITING(DOMEventTargetHelper)
 
 WorkerGlobalScopeBase::WorkerGlobalScopeBase(
-    NotNull<WorkerPrivate*> aWorkerPrivate,
-    UniquePtr<ClientSource> aClientSource)
+    WorkerPrivate* aWorkerPrivate, UniquePtr<ClientSource> aClientSource)
     : mWorkerPrivate(aWorkerPrivate),
       mClientSource(std::move(aClientSource)),
       mSerialEventTarget(aWorkerPrivate->HybridEventTarget()) {
@@ -762,9 +761,9 @@
 }
 
 DedicatedWorkerGlobalScope::DedicatedWorkerGlobalScope(
-    NotNull<WorkerPrivate*> aWorkerPrivate,
-    UniquePtr<ClientSource> aClientSource, const nsString& aName)
-    : WorkerGlobalScope(aWorkerPrivate, std::move(aClientSource)),
+    WorkerPrivate* aWorkerPrivate, UniquePtr<ClientSource> aClientSource,
+    const nsString& aName)
+    : WorkerGlobalScope(std::move(aWorkerPrivate), std::move(aClientSource)),
       NamedWorkerGlobalScopeMixin(aName) {}
 
 bool DedicatedWorkerGlobalScope::WrapGlobalObject(
@@ -814,9 +813,9 @@
 }
 
 SharedWorkerGlobalScope::SharedWorkerGlobalScope(
-    NotNull<WorkerPrivate*> aWorkerPrivate,
-    UniquePtr<ClientSource> aClientSource, const nsString& aName)
-    : WorkerGlobalScope(aWorkerPrivate, std::move(aClientSource)),
+    WorkerPrivate* aWorkerPrivate, UniquePtr<ClientSource> aClientSource,
+    const nsString& aName)
+    : WorkerGlobalScope(std::move(aWorkerPrivate), std::move(aClientSource)),
       NamedWorkerGlobalScopeMixin(aName) {}
 
 bool SharedWorkerGlobalScope::WrapGlobalObject(
@@ -848,10 +847,9 @@
 NS_IMPL_RELEASE_INHERITED(ServiceWorkerGlobalScope, WorkerGlobalScope)
 
 ServiceWorkerGlobalScope::ServiceWorkerGlobalScope(
-    NotNull<WorkerPrivate*> aWorkerPrivate,
-    UniquePtr<ClientSource> aClientSource,
+    WorkerPrivate* aWorkerPrivate, UniquePtr<ClientSource> aClientSource,
     const ServiceWorkerRegistrationDescriptor& aRegistrationDescriptor)
-    : WorkerGlobalScope(aWorkerPrivate, std::move(aClientSource)),
+    : WorkerGlobalScope(std::move(aWorkerPrivate), std::move(aClientSource)),
       mScope(NS_ConvertUTF8toUTF16(aRegistrationDescriptor.Scope()))
 
       // Eagerly create the registration because we will need to receive